src/Compilers/CSharp/Portable/Binder/ContextualAttributeBinder.cs

```diff
@@ -64,5 +66,10 @@ internal Symbol AttributeTarget
                 return _attributeTarget;
             }
         }
+
+        protected override CSDiagnosticInfo SingleTypeNameNotFound(string whereText, DiagnosticBag diagnostics, Location location, AssemblySymbol forwardedToAssembly)
```

Comment:
For example, try using ```typeof(UndefinedType)``` as an argument for an attribute. Error reported shouldn't imply that a type named "UndefinedTypeAttribute" is a suitable candidate.